c++/src/capnp/capability.h

```diff
@@ -737,6 +740,19 @@ struct List<T, Kind::INTERFACE> {
 // =======================================================================================
 // Inline implementation details
 
+template <typename T>
+RemotePromise<T> RemotePromise<T>::reducePromise(kj::Promise<RemotePromise>&& promise) {
+  kj::Tuple<kj::Promise<Response<T>>, kj::Promise<kj::Own<PipelineHook>>> splitPromise =
+      promise.then([](RemotePromise&& inner) {
+    return kj::tuple(kj::Promise<Response<T>>(kj::mv(inner)),
+                     PipelineHook::from(kj::mv(inner)));
```

Comment:
The double appearance of `kj::mv(inner)` here and on the previous line looks fishy to me. At first glance, I would guess that `inner` is invalid after the first move. Is multiple inheritance guaranteed to do the right thing here? (I.e. first move just takes the `Promise` part, and second move just takes the `Pipeline` part.)

Reply:
I think it's worth a code comment. :) I had to go and read the relevant recipients to convince myself this was okay, too.